app/services/omniauth/authentication_service.rb

```diff
@@ -3,9 +3,12 @@ class AuthenticationService
     delegate :authenticated?, to: :policy
 
     def initialize(params)
+      @params       = params
       @uid          = params[:uid]
       @provider     = params[:provider]
-      @first_name, @last_name, @email = params.fetch(:info, {}).values_at(:first_name, :last_name, :email)
+      @first_name   = params_info[:first_name] || twitter_name&.first
+      @last_name    = params_info[:last_name] || twitter_name&.last
```

Comment:
How about resolving `first_name, last_name & email` in a separate class based on provider?
It looks okay at the moment but it'll get really messy once we start adding more providers